### PR TITLE
chore: converted install command to typescript

### PR DESCRIPTION
**Pull request by Quadriphobs1**
Summary:
---------

Part of #683
Converted commands/install/* to typescript

Test Plan:
yarn test

### CODE REVIEW

**Comment by thymikee** [comment]
Can you rebase and make sure lint is passing? :)

File: packages/cli/src/commands/install/install.ts [hunk 1]
```diff
@@ -4,16 +4,17 @@
  * This source code is licensed under the MIT license found in the
  * LICENSE file in the root directory of this source tree.
  *
- * @flow
  */
-
-import type {ConfigT} from 'types';
+/* eslint-disable import/namespace, import/default */
```
Comment (thymikee) [inline packages/cli/src/commands/install/install.ts]: Please don't do this for entire file, just where it's needed. We can remove these inline later when cleaning up

File: packages/cli/src/commands/install/install.ts [hunk 2]
```diff
@@ -4,16 +4,17 @@
...
 import {logger} from '@react-native-community/cli-tools';
+// @ts-ignore FIXME after converting tools/packageManager
 import * as PackageManager from '../../tools/packageManager';
```
Comment (thymikee) [inline packages/cli/src/commands/install/install.ts]: packageManager is already converted, you need to rebase

File: packages/cli/src/commands/install/uninstall.ts [hunk 3]
```diff
@@ -4,15 +4,16 @@
  * This source code is licensed under the MIT license found in the
  * LICENSE file in the root directory of this source tree.
  *
- * @flow
  */
-
-import type {ConfigT} from 'types';
+/* eslint-disable import/namespace, import/default */
```
Comment (thymikee) [inline packages/cli/src/commands/install/uninstall.ts]: same as above

File: packages/cli/src/commands/install/uninstall.ts [hunk 4]
```diff
@@ -4,15 +4,16 @@
...
+import {Config} from '@react-native-community/cli-types';
 import {logger} from '@react-native-community/cli-tools';
+// @ts-ignore FIXME after converting tools/packageManager
 import * as PackageManager from '../../tools/packageManager';
```
Comment (thymikee) [inline packages/cli/src/commands/install/uninstall.ts]: packageManager is already converted

**Comment by Quadriphobs1** [comment]
@thymikee  Can you check it now

File: packages/cli/src/commands/install/install.ts (hunk 1 above)
Comment (thymikee) [reply packages/cli/src/commands/install/install.ts]: still there, please see my message above

**Comment by thymikee** [comment]
The issues I brought are still there, please address them first.

File: packages/cli/src/commands/install/install.ts (hunk 1 above)
Comment (Quadriphobs1) [reply packages/cli/src/commands/install/install.ts]: Removing it won't pass the test and the file `link/unlink` is not typed which is causing `import/namespace` typescript error

Comment (thymikee) [reply packages/cli/src/commands/install/install.ts]: All I ask for is to disable only the lines that need it. It's easy to forget about global disables, that's why we avoid them.

Comment (Quadriphobs1) [reply packages/cli/src/commands/install/install.ts]: +1

**Comment by thymikee** [comment]
Thanks!

**Comment by DanielUche** [comment]
wow some bit me to this task. I can take up any other task if any is available

**Comment by thymikee** [comment]
@DanielUche sorry, I didn't notice it was assigned to somebody else. Currently we don't have any TS-related task left. But you're free to help us with regular issues and code reviews :)

**Pull request merged**